### PR TITLE
CTCP1012 Expect x-message-id and improve error reporting if not present

### PR DESCRIPTION
**Pull request by ardwalker**

### CODE REVIEW

File: app/v2/services/InboundRouterService.scala [hunk 1]
```diff
@@ -53,9 +53,9 @@ class InboundRouterServiceImpl @Inject()(inboundRouterConnector: InboundRouterCo
         .post(messageType, message, departureId)
         .map(response => {
           if (is2xx(response.status)) {
-            response.header(LOCATION) match {
+            response.header(MessageIdHeaderKey) match {
               case Some(loc) => Right(MessageId(Utils.lastFragment(loc)))
-              case _         => Left(PersistenceError.UnexpectedError(None))
+              case _         => Left(PersistenceError.UnexpectedError(Some(new Exception("Location header missing from router response"))))
```
Comment (liandra-giacomine) [inline app/v2/services/InboundRouterService.scala:58]: the error message refers to the location header but the header we check is not the messageId header?

Comment (ardwalker) [reply app/v2/services/InboundRouterService.scala:58]: Forgot to rename loc variable, its reading the correct value from the header though

File: app/v2/services/InboundRouterService.scala [hunk 2]
```diff
@@ -53,9 +53,9 @@ class InboundRouterServiceImpl @Inject()(inboundRouterConnector: InboundRouterCo
         .post(messageType, message, departureId)
         .map(response => {
           if (is2xx(response.status)) {
-            response.header(LOCATION) match {
-              case Some(loc) => Right(MessageId(Utils.lastFragment(loc)))
-              case _         => Left(PersistenceError.UnexpectedError(None))
+            response.header(MessageIdHeaderKey) match {
+              case Some(value) => Right(MessageId(Utils.lastFragment(value)))
+              case _           => Left(PersistenceError.UnexpectedError(Some(new Exception("Location header missing from router response"))))
```
Comment (liandra-giacomine) [inline app/v2/services/InboundRouterService.scala:58]:
```suggestion
              case _           => Left(PersistenceError.UnexpectedError(Some(new Exception("X-Message-Id header missing from router response"))))
```

Comment (liandra-giacomine) [reply app/v2/services/InboundRouterService.scala:58]: Still refers to the location header

**Review by liandra-giacomine — DISMISSED** [review]

File: app/v2/generators/DepartureMessageGenerator.scala [hunk 3]
```diff
@@ -38,53 +37,49 @@ class DepartureMessageGenerator @Inject()(clock: Clock) extends Generator(clock)
...
-          <Address>
-            <streetAndNumber>{Strings.alphanumeric(8, 70)}</streetAndNumber>
-            <city>{Strings.alphanumeric(3, 35)}</city>
-            <postCode>{Strings.alphanumeric(6, 17)}</postCode>
-            <country>{Strings.alpha(2)}</country>
-          </Address>
-        </HolderOfTheTransitProcedure>
-      </CC928C>
-    </TraderChannelResponse>
+    <ncts:CC928C xmlns:ncts="http://ncts.dgtaxud.ec" PhaseID="NCTS5.0">
```
Comment (derekspalmer) [inline app/v2/generators/DepartureMessageGenerator.scala:40]: Why no TraderChannelResponse ?

Comment (derekspalmer) [reply app/v2/generators/DepartureMessageGenerator.scala:40]: Ok. Seen wrap()

**Review by derekspalmer — APPROVED** [review]

**Review by RobertLaverick — COMMENTED** [review]
Not sure I'm a huge fan of the raw NodeSeq being passed around - maybe have it be a Message type that can be extended with WrappedMessage?

**Review by RobertLaverick — APPROVED** [review]

**Pull request merged**